### ISSUE TITLE
Enable configuration gets provides from the outside during 'docker run'

### ISSUE THREAD

**Issue by k00ni** [Dockerizing/IGUANA#1]
### Own query file and configuration file

Make it possible to call the container using something like 

```
docker run \
    -v $PWD/iguana-config:/iguana/config \
    -v /tmp/iguana-results:/iguana/results \
    -it iguana:latest /bin/bash
```

so that the user is able to override the internal configuration of IGUANA by his own configuration file and query file.
### Auto execution

Furthermore, it should be possible to provide your configuration to the container, run it, let it create the results and than it exits on it own. So it acts more like an executable.

**Comment by simeonackermann** [comment]
With IGUANA.sh from 12a3da2f28d7e3397646622e91db16b18c228706 it should possibly to auto execute the docker environment with config, queries and results volumes.

To see all output messages it's required to start it from terminal. Currently the config files are hardcoded vars.

Do you think it's required that users can pass the paths to config files/result dir in a more comfortable way than as parameter?

**Comment by k00ni** [comment]
Hallo @simeonackermann,

thanks for the quick response and work you have done so far! 

But i think there was a misunderstanding between us. The objective here is to change the Docker container (resp. the Dockerfile and related files) in a way, so that, when the user deploys the container, he is able to override the internal configuration of IGUANA with his own configuration file, query file and folder path for results. Or he can choose to use these files inside the container (but the first option must be the standard way).

I want to lower the barrier and avoid as much hassle as possible. So you only have to pull the container and deploy it with your parameters. That's it. Further configuration may be done in the container itself, after it was started, but it should not concern the user. As i stated, think of this container as an executable with parameters given by the user.

Hope that descriptions made the objective more clear. If you have any questions dont hesitate to call or write me. 

Kinda regards

**Comment by k00ni** [comment]
I almost forgot the relevant part. Your last commit contains a shell script to setup the container. My problem with that is, that you need that file to work. But you cant "pull" that file like you can pull a docker container. So for now, i want to keep the IGUANA docker inside the Docker channels without any foreign entities like additional file. 

That does not mean, that you provide additional configuration or setup files, which will be executed inside the Docker container! Everything you need to accomplish something inside the container is fine, but it must not concern the user.

**Comment by simeonackermann** [comment]
Ok the script cannot pulled as a docker container. My intention was, that users just needs to download an execute that script (which pulls and runs the iguana container), without any docker pull/run commands.

If I got it right, the workflow should be firstly something like:
`docker pull aksw/dld-present-iguana`
`docker run -v... aksw/dld-present-iguana:latest`
which mounts a results and config directory?!

**Comment by k00ni** [comment]
Hey Simeon,

> My intention was, that users just needs to download an execute that script (which pulls and runs the iguana container), without any docker pull/run commands.

That approach is not wrong and might be also helpful, so lets keep that file and use it later on. But for now, lets concentrate on the straight docker way.

> If I got it right, the workflow should be firstly something like: 
> `docker pull aksw/dld-present-iguana`
> `docker run -v... aksw/dld-present-iguana:latest`
> which mounts a results and config directory?!

Yes, that is basically what i wrote in my introduction post. To my knowledge, you can not override only files inside the docker container while you are using `docker run` command. But you can override folders using `-v`, so my idea was, that you override the configuration folder of IGUANA inside the docker with a folder of your own, containing only the configuration files: queries.txt and config_test.xml. 

The question is, is there a way to override only the files? And how can we made it as easy as possible for the user, to provide the 2 configuration files and the results folder, when he uses `docker run` (to use the container basically like an executable). Your objective would be to be to solve these two questions, if possible.

Greetings

**Comment by simeonackermann** [comment]
Ok, we can [mount files](http://docs.docker.com/userguide/dockervolumes/#mount-a-host-file-as-a-data-volume) the same way as folders!

So the usage could be like:

```
docker run \
   -v $PWD/config.xml:/iguana/config_test.xml \
   -v $PWD/queries.txt:/iguana/queries.txt \
   -v $PWD/results:/iguana/results \ 
   iguana:latest
```

**Comment by k00ni** [comment]
Sounds great. Could you please adapt the README.md and replace the existing docker run command with yours? With that i think this issue is finished, isn't it?

**Comment by simeonackermann** [comment]
Closed this with e4bafa897ebee6ce19e82fad5d023ecc0236cc57
